src/app/clusters/time-synchronization-server/time-synchronization-cluster.cpp

```diff
@@ -925,377 +990,278 @@ TimeState TimeSynchronizationServer::UpdateDSTOffsetState()
-    }
-    case Granularity::Id: {
-        return aEncoder.Encode(TimeSynchronizationServer::Instance().GetGranularity());
-    }
-    case TrustedTimeSource::Id: {
-        return ReadTrustedTimeSource(aPath.mEndpointId, aEncoder);
-    }
-    case DefaultNTP::Id: {
-        return ReadDefaultNtp(aPath.mEndpointId, aEncoder);
-    }
-    case TimeZone::Id: {
-        return ReadTimeZone(aPath.mEndpointId, aEncoder);
-    }
-    case DSTOffset::Id: {
-        return ReadDSTOffset(aPath.mEndpointId, aEncoder);
-    }
-    case TimeZoneListMaxSize::Id: {
-        uint8_t max = CHIP_CONFIG_TIME_ZONE_LIST_MAX_SIZE;
-        return aEncoder.Encode(max);
-    }
-    case DSTOffsetListMaxSize::Id: {
-        uint8_t max = CHIP_CONFIG_DST_OFFSET_LIST_MAX_SIZE;
-        return aEncoder.Encode(max);
-    }
-    case LocalTime::Id: {
-        return ReadLocalTime(aPath.mEndpointId, aEncoder);
-    }
-    default: {
-        break;
-    }
-    }
-
-    return err;
+    return CHIP_NO_ERROR;
 }
-} // anonymous namespace
 
-bool emberAfTimeSynchronizationClusterSetUTCTimeCallback(
-    chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
-    const chip::app::Clusters::TimeSynchronization::Commands::SetUTCTime::DecodableType & commandData)
+Protocols::InteractionModel::Status
+TimeSynchronizationCluster::HandleSetUTCTime(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                             const Commands::SetUTCTime::DecodableType & commandData)
 {
     const auto & utcTime     = commandData.UTCTime;
     const auto & granularity = commandData.granularity;
     const auto & timeSource  = commandData.timeSource;
 
-    auto currentGranularity = TimeSynchronizationServer::Instance().GetGranularity();
+    auto currentGranularity = GetGranularity();
     if (granularity < GranularityEnum::kNoTimeGranularity || granularity > GranularityEnum::kMicrosecondsGranularity)
     {
-        commandObj->AddStatus(commandPath, Status::InvalidCommand);
-        return true;
+        return Protocols::InteractionModel::Status::InvalidCommand;
     }
     if (timeSource.HasValue() && (timeSource.Value() < TimeSourceEnum::kNone || timeSource.Value() > TimeSourceEnum::kGnss))
     {
-        commandObj->AddStatus(commandPath, Status::InvalidCommand);
-        return true;
+        return Protocols::InteractionModel::Status::InvalidCommand;
     }
 
     if (granularity != GranularityEnum::kNoTimeGranularity &&
         (currentGranularity == GranularityEnum::kNoTimeGranularity || granularity >= currentGranularity) &&
-        CHIP_NO_ERROR ==
-            TimeSynchronizationServer::Instance().SetUTCTime(commandPath.mEndpointId, utcTime, granularity, TimeSourceEnum::kAdmin))
+        CHIP_NO_ERROR == SetUTCTime(commandPath.mEndpointId, utcTime, granularity, TimeSourceEnum::kAdmin))
     {
-        commandObj->AddStatus(commandPath, Status::Success);
+        NotifyAttributeChanged(UTCTime::Id);
```

Comment:
Should this not be handled by `SetUTCTime` ?

Overall also the same comment regarding endpointId: we are supposedly a singleton on kRootEndpointId so we could omit this or always use mPath.mEndpointId if we really need it.